internal/tiltfile/tiltfile_state.go

```diff
@@ -308,6 +311,27 @@ func (s *tiltfileState) assembleK8sV2() error {
 		return err
 	}
 
+	for workload, opts := range s.k8sResourceOptions {
+		if r, ok := s.k8sByName[workload]; ok {
+			r.extraPodSelectors = opts.extraPodSelectors
+			r.portForwards = opts.portForwards
+			if opts.newName != "" && opts.newName != r.name {
+				if _, ok := s.k8sByName[opts.newName]; ok {
+					return fmt.Errorf("k8s_resource at %s specified to rename '%s' to '%s', but there is already a resource with that name", opts.tiltfilePosition.String(), r.name, opts.newName)
```

Comment:
can you add a test that exercises this error message?

Reply:
That's here: https://github.com/windmilleng/tilt/pull/1441/files#diff-61015eae0b8419289a8f2f1b45e156e6R2705